nixos/modules/services/web-servers/uwsgi.nix

```diff
@@ -0,0 +1,100 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.uwsgi;
+
+  python2Pkgs = pkgs.python2Packages.override {
+    python = pkgs.uwsgi.python2;
+    self = python2Pkgs;
+  };
+
+  python3Pkgs = pkgs.python3Packages.override {
+    python = pkgs.uwsgi.python3;
+    self = python3Pkgs;
+  };
+
+  buildCfg = n: c: if builtins.typeOf c != "set" then c else pkgs.writeText "${n}.json" (builtins.toJSON {
+    uwsgi =
+      if c.type == "normal"
+        then {
+          pythonpath =
+               (if c ? python2Packages
+                then builtins.map (x: "${x}/${pkgs.uwsgi.python2.sitePackages}") (c.python2Packages python2Pkgs)
+                else [])
+            ++ (if c ? python3Packages
+                then builtins.map (x: "${x}/${pkgs.uwsgi.python3.sitePackages}") (c.python3Packages python3Pkgs)
+                else []);
+          plugins = cfg.plugins;
+        } // removeAttrs c [ "type" "python2Packages" "python3Packages" ]
+      else if c.type == "emperor"
+        then {
+          emperor = if builtins.typeOf c.vassals != "set" then c.vassals
+                    else pkgs.buildEnv {
+                      name = "vassals";
+                      paths = mapAttrsToList buildCfg c.vassals;
+                    };
+        } // removeAttrs c [ "type" "vassals" ]
+      else abort "type should be either 'normal' or 'emperor'";
+  });
```

Comment:
Could you maybe reformat the expression here,...

I'd happily provide a patch for this, if you want me to... but I consider this as barely readable... sorry! :wink: